features/mesh/policies/Index.feature

```diff
@@ -92,3 +70,81 @@ Feature: mesh / policies / index
 
     Then the "$item:nth-child(1) td:nth-child(1)" element contains "mfi-1"
     And the "$item:nth-child(1)" element contains "MeshService:service-1"
+
+  Scenario: Hides legacy policy types if there are no legacy policies applied
+    Given the URL "/mesh-insights/default" responds with
+      """
+      body:
+        policies:
+          CircuitBreaker:
+            total: 0
+          FaultInjection:
+            total: 0
+          HealthChecks:
+            total: 0
+          MeshGatewayRoute:
+            total: 0
+          ProxyTemplate:
+            total: 0
+          RateLimit:
+            total: 0
+          Retry:
+            total: 0
+          Timeout:
+            total: 0
+          TrafficLog:
+            total: 0
+          TrafficPermission:
+            total: 0
+          TrafficRoute:
+            total: 0
+          TrafficTrace:
+            total: 0
+          VirtualOutbound:
+            total: 0
+          MeshFaultInjection:
+            total: 2
+      """
+
+    When I visit the "/meshes/default/policies/meshfaultinjections" URL
+
+    Then the "[data-testid^='policy-type-link-']" element exists 14 times
+
+  Scenario: Shows legacy policy types if there are any legacy policies applied
+    Given the URL "/mesh-insights/default" responds with
+      """
+      body:
+        policies:
+          CircuitBreaker:
+            total: 1
+          FaultInjection:
+            total: 0
+          HealthChecks:
+            total: 0
+          MeshGatewayRoute:
+            total: 0
+          ProxyTemplate:
+            total: 0
+          RateLimit:
+            total: 0
+          Retry:
+            total: 0
+          Timeout:
+            total: 0
+          TrafficLog:
+            total: 0
+          TrafficPermission:
+            total: 0
+          TrafficRoute:
+            total: 0
+          TrafficTrace:
+            total: 0
+          VirtualOutbound:
+            total: 0
+          MeshFaultInjection:
+            total: 2
+      """
+
+    When I visit the "/meshes/default/policies/meshfaultinjections" URL
+
+    Then the "[data-testid^='policy-type-link-']" element exists 27 times
```

Comment:
I kinda thought a lot about avoiding the counting and just setting one policy to 0 and looking for that, assuming that if one zero count legacy policy was showing then all the others are also. Then we wouldn't need to know that there are 27 that was have in our mocks. Its a bit of a quibble though so I don't mind really.

Reply:
I simplified this test as follows:

- In the second test, only setting one legacy policy type in the mocks
- In both tests, test against the presence (or lack of it) of a legacy policy type in the list